### ISSUE TITLE
Ability to disable calls to instance metadata endpoint 

### ISSUE THREAD

**Issue by jbrailsford** [microsoft/ApplicationInsights-Java#1775]
Currently when running the AI integration for Java for code that's deployed outside of Azure, we see repeated calls to the instance metadata service endpoint (http://169.254.169.254:80) - naturally these fail given it's not running in an environment with that available. 

Ideally it should be possible to disable this module in the same way it's possible to disable some of the automatic dependency telemtry modules.

If it's already possible to disable it, could it instead be documented?

**Comment by trask** [comment]
hi @jbrailsford! where are you seeing these calls? is the agent logging them, or are you capturing/seeing them some other way? thx

**Comment by jbrailsford** [comment]
Hey @trask, apologies for the delay, will confirm _what_ is logging them with our dev team tomorrow AM (UK time).

In the meantime, this may be of help, these are the log messages that are emitted:

```
2021-06-28 12:09:51.547Z INFO  o.a.http.impl.execchain.RetryExec - I/O exception (java.net.SocketException) caught when processing request to {}->http://169.254.169.254:80: Network is unreachable (connect failed)
2021-06-28 12:09:51.547Z INFO  o.a.http.impl.execchain.RetryExec - Retrying request to {}->http://169.254.169.254:80
2021-06-28 12:09:51.554Z INFO  o.a.http.impl.execchain.RetryExec - I/O exception (java.net.SocketException) caught when processing request to {}->http://169.254.169.254:80: Network is unreachable (connect failed)
…
```

**Comment by jbrailsford** [comment]
Back again, it appears that the exceptions are being logged by Log4J.

**Comment by trask** [comment]
hey @jbrailsford, sorry for the late reply, logging this at INFO was not intentional, and will be [fixed in the next release](https://github.com/microsoft/ApplicationInsights-Java/issues/1798#issuecomment-880890503) 👍

**Comment by jbrailsford** [comment]
hey @trask - that's great to hear, thanks for letting me know!

**Comment by trask** [comment]
This is fixed in `main` and will be in 3.2.0-BETA.2

**Comment by trask** [comment]
[3.2.0-BETA.2](https://github.com/microsoft/ApplicationInsights-Java/releases/tag/3.2.0-BETA.2) is released
